app/models_types.py

```diff
@@ -0,0 +1,349 @@
+    billing_contact_names: str | None
+    billing_contact_email_addresses: str | None
+    billing_reference: str | None
+    can_approve_own_go_live_requests: bool
+    permissions: list[str]
+
+
+class SerializedOrganisationForList(TypedDict):
+    name: str
+    id: str
+    active: bool
+    count_of_live_services: int
+    domains: list[str]
+    organisation_type: str | None
+
+
+class SerializedServiceOrgDashboard(TypedDict):
+    id: str
+    name: str
+    active: bool
+    restricted: bool
+
+
+class SerializedFreeSmsItems(TypedDict):
+    free_sms_fragment_limit: int
+    financial_year_start: int
+
+
+class SerializedService(TypedDict):
+    id: str
+    name: str
+
+
+class SerializedAnnualBilling(TypedDict):
+    id: str
+    free_sms_fragment_limit: int
+    service_id: str
+    financial_year_start: int
+    created_at: str
+    updated_at: str | None
+    service: SerializedService | None
+
+
+class SerializedInboundNumber(TypedDict):
+    id: str
+    number: str
+    provider: str
+    service: SerializedService | None
+    active: bool
+    created_at: str
+    updated_at: str | None
+
+
+class SerializedServiceSmsSender(TypedDict):
+    id: str
+    sms_sender: str
+    service_id: str
+    is_default: bool
+    archived: bool
+    inbound_number_id: str | None
+    created_at: str
+    updated_at: str | None
+
+
+class SerializedServiceCallbackApi(TypedDict):
+    id: str
+    service_id: str
+    url: str
+    updated_by_id: str
+    created_at: str
+    updated_at: str | None
+
+
+class SerializedTemplateFolder(TypedDict):
+    id: str
+    name: str
+    parent_id: str | None
+    service_id: str
+    users_with_permission: list[str]
+
+
+class SmsCostDetails(TypedDict):
+    billable_sms_fragments: int
+    international_rate_multiplier: float
+    sms_rate: float
+
+
+class LetterCostDetails(TypedDict):
+    billable_sheets_of_paper: int
+    postage: str
+
+
+class SerializedNotificationForCSV(TypedDict):
+    id: str
+    row_number: str
+    recipient: str
+    client_reference: str
+    template_name: str
+    template_type: str
+    job_name: str
+    status: str
+    created_at: str
+    created_by_name: str
+    created_by_email_address: str
+    api_key_name: str | None
+
+
+class SerializedNotification(TypedDict):
+    id: str
+    reference: str | None
+    email_address: str | None
+    phone_number: str | None
+    line_1: str | None
+    line_2: str | None
+    line_3: str | None
+    line_4: str | None
+    line_5: str | None
+    line_6: str | None
+    postcode: str | None
+    type: str
+    status: str
+    template: dict
+    body: str
+    subject: str | None
+    created_at: str
+    created_by_name: str | None
+    sent_at: str | None
+    completed_at: str | None
+    scheduled_for: str | None
+    postage: str | None
+    one_click_unsubscribe_url: str | None
+    estimated_delivery: str | None
+    cost_details: SmsCostDetails | LetterCostDetails | dict | None = None
```

Comment:
Having default values like this (and on the lines below) give the error of `Right hand side values are not supported in TypedDict` when running this through a static type checker. I haven't looked into the best way of fixing this, but there are some cases where the keys to do with cost shouldn't be in the dict at all, not just there with a default of None - not sure if that means we'll need two classes here or if there's a way to do that with one class

Reply:
This error came from Mypy, which I tried out with this branch

Reply:
Thanks - Fixed